Sources/Sentry/SentryScope.m

```diff
@@ -456,16 +447,31 @@ - (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
         event.level = level;
     }
 
-    id<SentrySpan> span;
-    @synchronized(_spanLock) {
-        span = self.span;
+    NSMutableDictionary *newContext;
+    if (nil == event.context) {
+        newContext = [self context].mutableCopy;
+    } else {
+        newContext = [NSMutableDictionary new];
+        [newContext addEntriesFromDictionary:[self context]];
+        [newContext addEntriesFromDictionary:event.context];
     }
 
-    if (![event.type isEqualToString:SentryEnvelopeItemTypeTransaction] &&
-        [span isKindOfClass:[SentryTracer class]]) {
-        event.transaction = [(SentryTracer *)span name];
-    }
+    if (self.span != nil) {
+        id<SentrySpan> span;
+        @synchronized(_spanLock) {
+            span = self.span;
+        }
 
+        // Span could be nil as we do the first check outside the synchronize
+        if (span != nil) {
+            if (![event.type isEqualToString:SentryEnvelopeItemTypeTransaction] &&
+                [span isKindOfClass:[SentryTracer class]]) {
+                event.transaction = [(SentryTracer *)span name];
+            }
+            newContext[@"trace"] = [span serialize];
```

Comment:
you want to serialize SentrySpanContext here, https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentrySpan.m#L20